dallinger/recruiters.py

```diff
@@ -77,6 +81,11 @@ class PsiTurkRecruiter(Recruiter):
 
     def __init__(self):
         """Set up the connection to MTurk and psiTurk web services."""
+
+        # commit to the db so if anything goes wrong the rest of the
+        # request is safe
+        session.commit()
```

Comment:
you are correct and yes we can do that, though it will require adding more commit() lines as it will have to be added every time the recruiter used after the db has been edited. For this reason I favor the current implementation, but would be happy to change it if you prefer, what do you think?